src/integrator_interface.jl

```diff
@@ -1,67 +1,102 @@
 function savevalues!(integrator::DDEIntegrator,force_save=false)
+  # update ODE integrator with values of DDE integrator
   integrator.integrator.u = integrator.u
   integrator.integrator.k = integrator.k
   integrator.integrator.t = integrator.t
+
+  # add steps for interpolation when needed
   OrdinaryDiffEq.ode_addsteps!(integrator.integrator,integrator.f)
+
+  # update solution of ODE integrator
   savevalues!(integrator.integrator,force_save)
 end
 
 function postamble!(integrator::DDEIntegrator)
+  # update ODE integrator with values of DDE integrator
   integrator.integrator.u = integrator.u
   integrator.integrator.k = integrator.k
   integrator.integrator.t = integrator.t
+
+  # clean up solution of ODE integrator
   OrdinaryDiffEq.postamble!(integrator.integrator)
 end
 
 function perform_step!(integrator::DDEIntegrator)
   integrator.tprev = integrator.t # this is necessary to extrapolate from the current interval
+
+  # update ODE integrator with values of DDE integrator
   integrator.integrator.uprev = integrator.uprev
   integrator.integrator.tprev = integrator.tprev
   integrator.integrator.fsalfirst = integrator.fsalfirst
   integrator.integrator.t = integrator.t
   integrator.integrator.dt = integrator.dt
 
-  # if dt>max lag, then it's explicit so use Picard iteration
+  # if dt>min lag, then it's explicit so use fixed-point iteration
   if integrator.dt >minimum(integrator.prob.lags)
-
-    # the is done to correct the extrapolation
-    t_prev_cache = integrator.tprev
+    # this is done to correct the extrapolation
     t_cache = integrator.t
-    uprev_cache = integrator.uprev
+    tprev_cache = integrator.tprev
+    if typeof(integrator.uprev_cache) <: AbstractArray
```

Comment:
Oh yes, my bad. So the question is rather: why is it saved in the first place? If entries of `integrator.uprev` are changed, entries of `uprev_cache` are also changed - so when we later set `integrator.uprev` to `uprev_cache` again, entries of `integrator.uprev` are in fact not updated to any prior values. Moreover, just by skimming through `low_order_rk_integrators.jl` in OrdinaryDiffEq.jl it seems that `integrator.uprev` is not updated by `perform_step!` - but I assume there are algorithms which also update `uprev`? Otherwise `uprev_cache` could be removed completely.